### PR TITLE
fix(clearnode): resolve a set of audit findings

### PR DESCRIPTION
**Pull request by philanton**
## Summary by CodeRabbit

* **Bug Fixes**
  * Wallet comparison is now case-insensitive to prevent false "same wallet" rejections.
  * Requests containing duplicate participant+asset allocations are rejected with clear error messaging.

* **New Features**
  * Validation of asset decimal precision for allocation amounts to enforce allowed decimals.

* **Tests**
  * Added tests covering duplicate-allocation rejection, deposit precision validation, rebalance duplicate handling, and same-wallet case-insensitive rejection.

### CODE REVIEW

File: clearnode/api/app_session_v1/submit_app_state_test.go [hunk 1]
```diff
@@ -1720,3 +1720,109 @@ func TestSubmitAppState_AppRegistryDisabled(t *testing.T) {
 	mockStore.AssertNotCalled(t, "GetApp", mock.Anything)
 	mockStore.AssertExpectations(t)
 }
+
+func TestSubmitAppState_OperateIntent_DuplicateAllocation_Rejected(t *testing.T) {
```
Comment (nksazonov) [inline clearnode/api/app_session_v1/submit_app_state_test.go:1907]: Please, add tests for rejecting duplicate allocations for all other app session endpoints; and for other issues fixed in this PR.

**Review by nksazonov — COMMENTED** [review]
LGTM, more tests needed

File: clearnode/api/channel_v1/submit_state_test.go [hunk 2]
```diff
@@ -344,6 +344,112 @@ func TestSubmitState_TransferSend_ReceiverWithEscrowLock_Rejected(t *testing.T)
...
+	}
+
+	// Derive senderWallet from a real key — Address().String() returns checksummed (mixed case)
+	userSigner := NewMockSigner()
+	userWalletSigner, _ := core.NewChannelDefaultSigner(userSigner)
+	senderWallet := userSigner.PublicKey().Address().String() // e.g. "0xAbCdEf..."
+
+	// Use the SAME address as receiver — the transition AccountID stores it as-is,
+	// but issueTransferReceiverState calls NormalizeHexAddress which lowercases it.
+	// The fix uses strings.EqualFold to catch this.
```
Comment (nksazonov) [inline clearnode/api/channel_v1/submit_state_test.go]: Wait. Does it mean that `issueTransferReceieverState` function itself introduces the bug of a lower-cased address? Why would it do that? Is it possible to remove such logic from it?

Also, if `issueTransferReceieverState` does lower-case the address, then if used directly, does it mean that the `strings.EqualFold` will always fail unless the address is checksummed-back after `issueTransferReceieverState`?

Comment (philanton) [reply clearnode/api/channel_v1/submit_state_test.go]: Inside our system, we use only lower-cased addresses right now (for storing and logging, so the data we browse has the same format everywhere). When the time comes, it would be good to move to just bytes. Comparison always uses `strings.EqualFold`, which is the alternative to converting to bytes and comparing them.

**Review by nksazonov — APPROVED** [review]
